apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+          }
+
+          var deltaElemsToStartScroll = 3;
+
+          this.elements[this.selectedElemIndex].classList.remove('selected');
+          if (this.selectedElemIndex >= 0 && this.selectedElemIndex < 4) {
+
+            cycleElements.call(this, (deltaElemsToStartScroll - this.selectedElemIndex));
+            this.selectedElemIndex += deltaElemsToStartScroll - this.selectedElemIndex;
+
+            this.selectedElemIndex--;
+            this.elements[this.selectedElemIndex].classList.add('selected');
+
+          } else {
+
+            this.selectedElemIndex--;
+            this.elements[this.selectedElemIndex].classList.add('selected');
+
+            if (this.selectedElemIndex > 3) {
+              this.elemList.style.top = parseFloat(this.elemList.style.top) + 6 + 'rem';
+              this.elemList.dataset.scrollup = this.elemList.style.top;
+            }
+
+          }
+
+          var lastElem = this.elements[this.elements.length - 1];
+          var clonedElem = lastElem.cloneNode(true);
+
+
+          this.elemList.insertBefore(clonedElem, this.elements[0]);
+          this.elements.splice(0, 0, clonedElem);
+          this.elements[0].classList.add('out-focus');
+          this.selectedElemIndex++;
+
+          lastElem.classList.add('removed');
+          this.elements.splice(-1, 1);
+
+          this.prevVerticalKey = 'up';
+
+
+          break;
+
+        case 'ArrowDown':
+
+          preVerticalNavigation.call(this);
+          document.getElementById('clock').style.opacity = 0;
+
+          this.elements[this.selectedElemIndex].classList.remove('selected');
+          if (this.selectedElemIndex == (this.elements.length - 3)) {
+            if (!this.elemList.dataset.scrolldown) {
+              this.elemList.style.top = parseFloat(this.elemList.style.top) - 6 + 'rem';
+              this.elemList.dataset.scrolldown = this.elemList.style.top;
+            }
+
+            this.selectedElemIndex--;
+
+            var removed = this.elements.splice(0, 1);
+
+            var cloned = removed[0].cloneNode(true);
+
+            removed[0].classList.add('removed');
+
+            this.elemList.appendChild(cloned);
+            this.elements[this.elements.length] = cloned;
+
+            this.selectedElemIndex++;
+            this.elements[this.selectedElemIndex].classList.add('selected');
+
+            this.elemList.style.top = this.elemList.dataset.scrolldown;
+          } else {
+            this.selectedElemIndex++;
+            this.elements[this.selectedElemIndex].classList.add('selected');
+
+            this.elemList.style.top = parseFloat(this.elemList.style.top) - 6 + 'rem';
+            this.elemList.dataset.scrolldown = this.elemList.style.top;
+          }
+
+          this.prevVerticalKey = 'down';
+
+          if (this.selectedElemIndex == 1) {
+            this.elemList.style.top = '0rem';
+          }
+
+          break;
+
+        case 'Accept':
+
+          window.removeEventListener('keydown', this);
```

Comment:
It doesn't seem like you should have to do this.  If you launch an app, then that app will get focus and we'll lose it, so the handler can remain registered.  In the case where you're just showing the more apps, maybe you can just check a flag to make sure we're not showing more apps before handling the key event here...  It seems like your code would be simpler if you didn't have to keep adding and removing this handler.